components/brave_wallet_ui/components/desktop/views/portfolio/index.tsx

```diff
@@ -42,98 +98,236 @@ const Portfolio = () => {
+  }, [selectedAsset])
+
+  // This will return the name of an account from user preferences
+  const accountName = (asset: RPCResponseType) => {
+    const foundName = mockUserAccounts.find((account) => account.address === asset.address)?.name
+    const name = foundName ? foundName : locale.account
+    return name
   }
 
+  // This filters a list of assets when the user types in search bar
   const filterAssets = (event: any) => {
     const search = event.target.value
     if (search === '') {
-      setAssetList(mockRPCResponse[0].assets)
+      setUserAssetList(createdAssetList)
     } else {
-      const filteredList = mockRPCResponse[0].assets.filter((asset) => {
+      const filteredList = createdAssetList.filter((asset) => {
         return (
           asset.name.toLowerCase() === search.toLowerCase() ||
           asset.name.toLowerCase().startsWith(search.toLowerCase()) ||
           asset.symbol.toLocaleLowerCase() === search.toLowerCase() ||
           asset.symbol.toLowerCase().startsWith(search.toLowerCase())
         )
       })
-      setAssetList(filteredList)
+      setUserAssetList(filteredList)
     }
   }
 
-  const expandTo = (asset: string) => () => {
-    alert(`Will Expand to ${asset} view`)
+  // This returns a list of transactions from all accounts filtered by selected asset
+  const transactions = React.useMemo(() => {
+    const response = mockRPCResponse
+    const transactionList = response.map((account) => {
+      const id = selectedAsset ? selectedAsset.id : ''
+      return account.transactions.find((item) => item.assetId === id)
+    })
+    const removedEmptyTransactions = transactionList.filter(x => x)
+    return removedEmptyTransactions
+  }, [selectedAsset])
+
+  // This will scrape all of the user's accounts and combine the balances for a single asset
+  const scrapedFullAssetBalance = (asset: AssetOptionType) => {
+    const response = mockRPCResponse
+    const amounts = response.map((account) => {
+      const balance = account.assets.find((item) => item.id === asset.id)?.balance
+      return balance ? balance : 0
+    })
+    const grandTotal = amounts.reduce(function (a, b) {
+      return a + b
+    }, 0)
+    return grandTotal
+  }
+
+  // This will scrape all of the user's accounts and combine the fiat value for a single asset
+  const scrapedFullAssetFiatBalance = (asset: AssetOptionType) => {
+    const fullBallance = scrapedFullAssetBalance(asset)
+    const price = CurrentPriceMockData.find((coin) => coin.symbol === asset?.symbol)?.usd
+    const value = price ? price * fullBallance : 0
+    return value
+  }
+
+  // This will scrape all of the user's accounts and combine the fiat value for every asset
+  const scrapedFullPortfolioBalance = () => {
+    const amountList = createdAssetList.map((asset) => {
+      return scrapedFullAssetFiatBalance(asset)
+    })
+    const grandTotal = amountList.reduce(function (a, b) {
+      return a + b
+    }, 0)
+    return formatePrices(grandTotal)
```

Comment:
I know this is all mock data but just want to confirm that we won't have all of this logic inside the UI component, and put inside a redux reducer or redux async handler, or something similar? We could also do this for the mock data (have a Storybook state / reducer module), but it's fine here for now also. It just might make integration easier if this UI module simply accepts the props and actions it needs, and then we fill in the missing pieces elsewhere later.

Reply:
Yes sir, that is the plan to use reducers in the future! Will implement that soon :)